server/sonar-web/src/main/js/apps/overview/events/PreviewGraph.js

```diff
@@ -139,44 +141,49 @@ export default class PreviewGraph extends React.PureComponent {
 
   render() {
     const { graph, selectedDate, series, tooltipIdx, tooltipXPos } = this.state;
+
     return (
       <div
         className="overview-analysis-graph big-spacer-bottom spacer-top"
         onClick={this.handleClick}
         tabIndex={0}
         role="link">
-        <AutoSizer disableHeight={true}>
-          {({ width }) =>
-            <div>
-              <AdvancedTimeline
-                endDate={null}
-                startDate={null}
-                height={80}
-                width={width}
-                hideGrid={true}
-                hideXAxis={true}
-                interpolate="linear"
-                metricType={getSeriesMetricType(series)}
-                padding={GRAPH_PADDING}
-                series={series}
-                showAreas={['coverage', 'duplications'].includes(graph)}
-                updateTooltip={this.updateTooltip}
-              />
-              {selectedDate != null &&
-                tooltipXPos != null &&
-                tooltipIdx != null &&
-                <PreviewGraphTooltips
-                  formatValue={this.formatValue}
-                  graph={graph}
-                  graphWidth={width}
-                  metrics={this.props.metrics}
-                  selectedDate={selectedDate}
-                  series={series}
-                  tooltipIdx={tooltipIdx}
-                  tooltipPos={tooltipXPos}
-                />}
+        {hasHistoryDataValue(series)
+          ? <AutoSizer disableHeight={true}>
```

Comment:
Could you extract it to a dedicated function please (like `renderTimeline`)? It'd increase the readability.